### ISSUE TITLE
Restore doesn't work 

### ISSUE THREAD

**Issue by djhaskin987** [spersson/Kup#24]
Whether I choose "Restore to original location" or try to browse to restore location, I hit the button and nothing happens.

**Comment by spersson** [comment]
Can not reproduce. Please provide more details if you can, 
1. Perhaps there is some error output that can be helpful, try running "kup-filedigger" from konsole, give it the path of your backups as the only argument. See if there are any interesting errors shown in the output when trying to restore.
2. Try to see if it works under any other conditions, with a new backup plan or different settings. restoring a file instead of a folder, etc... 

Thank you for reporting!

**Comment by djhaskin987** [comment]
K, here's what I found.

When I'm restoring an individual file, or a directory *with no
subdirectories*, it works. It's when it is restoring a whole directory
tree with files in it that it doesn't get past next.

kup-filedigger yields no errors when I hit "next".

It doesn't matter what files I'm trying to restore; it's not because the
directory I am restoring is special or even that it has symbolic links.
I choose to restore a folder with subfolders, and hit next, then select
where to restore, then hit next, and no worky :(

SO SAD I love this tool.

**Comment by spersson** [comment]
I had a look at the code for restoring now... Found that there is one case where could stop, not do anything, and not tell you about it. I had forgotten to add the code for displaying the error message. Let's hope that is what you see. You can test by compiling the latest code here. Or just wait a bit, I'm planning for a bug fix release very soon, like this week or so. But yeah, this would only help by showing some error text explaining what is going on (why it is failing to get a list of all files that are about to be restored).

You can also test already now with your current installation, simply click "Open" (next to the Restore button) on a folder in filedigger. Check that a file manager opens and shows you the contents of that folder in your backup archive. If that works then i still don't know what is going on in your case. If it doesn't work then it would be helpful to see the error message you get.

**Comment by Saboteur777** [comment]
I faced this issue, too. Just in case I tried to restore some folders with kup-filedigger. Some of them worked, but some of them failed. I ran kup-filedigger from terminal, got the following output: 

[saboteur@XPS:~]$ kup-filedigger /run/media/saboteur/a84cdd85-afb9-4bb4-a885-b7b4054601ea/Biztonsági\ mentések/
Icon theme "ubuntu-mono-dark" not found.
Icon theme "Mint-X" not found.
Icon theme "elementary" not found.
kf5.kio.core: We got some errors while running testparm "Load smb config files from /etc/samba/smb.conf\nError loading services.\n"
kf5.kio.core: KSambaShare: Could not find smb.conf!
kf5.kio.core: We got some errors while running 'net usershare info'
kf5.kio.core: "Can't load /etc/samba/smb.conf - run testparm to debug it\n"
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 36234 bytes
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38902 bytes
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes                                                   
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes                                                   
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes                                                   
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes                                                   
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes                                                   
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes                                                   
kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of 38916 bytes  

I received the `kf5.kio.core: KLocalSocket(0xb99ce73bf0) Jumbo packet of` messages when clicking on the Next button, after choosing where to restore.

**Comment by spersson** [comment]
Searching around based on your input, I found this one: [https://bbs.archlinux.org/viewtopic.php?id=201599](https://bbs.archlinux.org/viewtopic.php?id=201599). I agree that the warnings about jumbo packets seems pointless, does not look like that is a problem. But I'm thinking if you could be seeing the same problem as reported there... that an invisible dialog is opened (I didn't know that was possible!) and that just pressing Esc will make the process continue. Can you try that?

**Comment by Saboteur777** [comment]
I made a copy of my `/etc/samba/smb.conf.default` to `/etc/samba/smb.conf`, so the SMB warning (`We got some errors while running testparm "Load smb config files from /etc/samba/smb.conf\nWARNING: No path in service printers - making it unavailable!\nLoaded services file OK.\n"
`) is gone, but the jumbo pocket problem remained. 

I tried to press ESC after clicking on next, but that just closed the dialog. I made a video about this - sorry for the bad quality: https://streamable.com/mmjx5

**Comment by spersson** [comment]
@Saboteur777 do you still see this problem? I never managed to get anywhere in trying to figure out what happens.

**Comment by stefano-b** [comment]
I'm having problems to restore files too.
Like others have reported, individual files or directories containing only files are restored normally, the problem arises when trying to restore a directory that contains other directories too.

This is a serious issue, if you need some other feedback please ask.

**Comment by djhaskin987** [comment]
Just wanted to say @spersson this is a great tool, thanks for working on it.

**Comment by spersson** [comment]
@stefano-b I'm sorry to say that I still can't reproduce the issue. Just to rule out if there is something with your backup or with the filesystem that bup can't handle... can you run the restore command that kup runs and see if that works? Use the following command: bup -d /path/to/repo restore -vv -C /where/to/restore /what/to/restore

**Comment by stefano-b** [comment]
@spersson thank you for your reply.

I've tried and got some errors, here is the output:

`~> bup -d /run/media/stefano/DATA/backup/sbweb-LAMP/ restore -vv -C /home/sbweb/Scrivania/restore/ *
Warning: python-xattr module is too old; install python-pyxattr instead.
ERROR: path 'Apps' doesn't include a branch and revision              
ERROR: path 'Documenti' doesn't include a branch and revision         
ERROR: path 'Ebook' doesn't include a branch and revision             
ERROR: path 'Font' doesn't include a branch and revision              
ERROR: path 'Immagini' doesn't include a branch and revision          
ERROR: path 'Lavori' doesn't include a branch and revision            
ERROR: path 'Modelli' doesn't include a branch and revision           
ERROR: path 'Pubblici' doesn't include a branch and revision          
ERROR: path 'Scaricati' doesn't include a branch and revision         
ERROR: path 'Scripts' doesn't include a branch and revision           
ERROR: path 'Scrivania' doesn't include a branch and revision         
Restoring: 0, done.
WARNING: 11 errors encountered while restoring.`

If it matters, the repo is on a NTFS partition.

**Comment by spersson** [comment]
Sorry, I was a bit rushed when I wrote last comment. The part of "/what/to/restore" was not helpful. Actually you need to specify a path similar to this: "kup/latest/". That would restore the complete last captured backup. In my case I did this test just now to restore a single file:
```
bup -d ~/test_dest/ restore -vv -C /home/simon/wheee kup/latest/home/simon/test/test.7z
```
You can see paths available to be restored by using 
```
bup -d /path/to/repo ls
```

**Comment by stefano-b** [comment]
Thank you for your help, by specifying a path like you pointed out, the restore process went fine. I guess that we can rule out problems with the backup repository.

**Comment by spersson** [comment]
I just spent an hour reading through the kup code for restoring... still can't think of anything that could cause this. I'm missing something. Can you confirm that clicking "Open" instead of "Restore" for a folder works OK? I just want to know that the bup:// protocol handler works correctly for the folder you are trying to restore. Just see that a file manager opens up and you can see all the files and folders in the backup.
Next thing I want to confirm is exactly what you see happening in the restore dialog.  After clicking "Choose where to restore", then selecting a folder and clicking "Next". Does the window change at all? If I understood the original reporter then absolutely nothing changes on screen. At this point, could you check the process monitor by pressing ctrl+esc. There you select in top right corner to show All processes, Tree. Then find kup-filedigger and see if it is running any sub-process?
Thanks for your help so far. It's tricky to do troubleshooting when I don't have access to see the problem myself. So far I have never really had to deal much with such a situation.

**Comment by spersson** [comment]
I am working on adding diagnostics to kup, will be part of next release.
In the mean time, perhaps you can run the following command from a bash shell?

QT_LOGGING_RULES="kup*=true;kf5.kio*=true" kup-filedigger /path/to/your/backups  2> test-results

Then post the test-results file here. Thanks!

**Comment by stefano-b** [comment]
This is the output.

```
Icon theme "elementary" not found.
Icon theme "gnome" not found.
Icon theme "deepin" not found.
Icon theme "elementary" not found.
Icon theme "gnome" not found.
kf5.kio.core: "/usr/lib64/qt5/plugins/kf5/kio/gdrive.so" supports protocols ("gdrive")
kf5.kio.core: "/usr/lib64/qt5/plugins/kf5/kio/zeroconf.so" supports protocols ("zeroconf")
kf5.kio.core: "/usr/lib64/qt5/plugins/kf5/kio/kdeconnect.so" supports protocols ("kdeconnect")
kf5.kio.core: "/usr/lib64/qt5/plugins/kf5/kio/file.so" supports protocols ("file")
kf5.kio.core: "/usr/lib64/qt5/plugins/kf5/kio/ftp.so" supports protocols ("ftp")
kf5.kio.core: "/usr/lib64/qt5/plugins/kf5/kio/ghelp.so" supports protocols ("ghelp")
kf5.kio.core: "/usr/lib64/qt5/plugins/kf5/kio/help.so" supports protocols ("help")
kf5.kio.core: "/usr/lib64/qt5/plugins/kf5/kio/http.so" supports protocols ("http", "https", "webdav", "webdavs")
kf5.kio.core: "/usr/lib64/qt5/plugins/kf5/kio/trash.so" supports protocols ("trash")
kf5.kio.core: "/usr/lib64/qt5/plugins/kf5/kio/remote.so" supports protocols ("remote")
```

**Comment by spersson** [comment]
I finally found some time and motivation for looking at this one again. Best I can think of is that the source file scanning takes long and that's why it appears that nothing is happening. I have added progress monitoring for this operation now. And fixed hiding the empty message bar. And enabled debug output even in release builds.

[commit](https://invent.kde.org/kde/kup/-/commit/e720ac455a60bb3d33ef3e82f26caf82b10c3a88
)
